packages/Avalonia/AvaloniaRules.Project.xml

```diff
@@ -0,0 +1,81 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Rule Name="Avalonia" DisplayName="Avalonia" Order="70" PageTemplate="generic" xmlns="http://schemas.microsoft.com/build/2009/properties">
+
+  <Rule.Categories>
+    <Category Name="Compiling" DisplayName="Compiling" />
+    <Category Name="Debug" DisplayName="Debugging" />
+    <Category Name="NameGenerator" DisplayName="Nameming Generator" />
+  </Rule.Categories>
+
+  <Rule.DataSource>
+    <DataSource Persistence="ProjectFile" HasConfigurationCondition="false" Label="Avalonia" />
+  </Rule.DataSource>
+
+  <!-- Compiling -->
+  <BoolProperty Name="AvaloniaUseExternalMSBuild"
+                DisplayName="External MSBuild"
+                Description="Use dotnet build instead of MSBuild"
+                Category="Compiling" />
+
+  <BoolProperty Name="AvaloniaUseCompiledBindingsByDefault"
+              DisplayName="Use CompiledBindings"
+              Description="Use by default CompiledBindings"
+              Category="Compiling" />
+
+  <BoolProperty Name="AvaloniaAccessUnstablePrivateApis"
+            DisplayName="Use Private Apis"
+            Description="Enable access to unstable private APIs (not recommended)"
+            Category="Compiling" />
+
+  <!-- Debug -->
+
+  <BoolProperty Name="AvaloniaXamlIlDebuggerLaunch"
+          DisplayName="Debug Xaml Compaliler"
+          Description="Allow debug Xaml compilation."
+          Category="Debug" />
+
+
+  <EnumProperty Name="AvaloniaXamlReportImportance"
+                DisplayName="Xaml Report Importance"
+                Description="Provides levels of importance for Xaml Compiler messages"
+                Category="Debug">
+    <EnumValue Name="Low" DisplayName="Low" Description="Set Xaml Compiler report message to Low importance" />
+    <EnumValue Name="Normal" DisplayName="Normal" Description="Set Xaml Compiler report message to Normal importance" />
+    <EnumValue Name="High" DisplayName="High" Description="Set Xaml Compiler report message to High importance" />
+  </EnumProperty>
+
+  <!-- NameGenerator -->
+
+  <BoolProperty Name="AvaloniaNameGeneratorIsEnabled"
+              DisplayName="Use Name Generator"
+              Description="Enable/Disable Avalalina Name Source Generator"
+              Category="NameGenerator" />
+
+  <EnumProperty Name="AvaloniaNameGeneratorBehavior"
+                DisplayName="Generator Behavior"
+                Description="Customize behavior of name generator"
+                Category="NameGenerator">
+    <EnumValue Name="InitializeComponent" DisplayName="InitializeComponent" Description="The binding of the name to the control is done only once in InitializeComponent and is stored in a field." />
+    <EnumValue Name="OnlyProperties" DisplayName="OnlyProperties" Description="Generator generate a Properties to binding name to control" />
```

Comment:
The descriptions are wrong here I think. https://github.com/AvaloniaUI/Avalonia.NameGenerator#advanced-usage

InitializeComponent:
"Generates get-only properties to access controls by name and an InitializeComponent method to set them automatically. This is closes to other XAML UI framework functionality."

OnlyProperties:
"Generates get-only properties to access controls by name but does not generate an InitializeComponent method to set them automatically. This allows custom InitializeComponent method implementations"